plugins/org.yakindu.sct.simulation.ui/src/org/yakindu/sct/simulation/ui/view/ExecutionContextLabelProvider.java

```diff
@@ -113,4 +148,108 @@ private void updateNameCell(ViewerCell cell) {
 		}
 	}
 
+	/**
+	 * 
+	 * @author robert rudi - Initial contribution and API
+	 * 
+	 */
+	protected final class TreeEditorDisposeListener implements DisposeListener {
+
+		private static final String LISTENER_DATA = "DISPOSELISTENER";
+		private static final String EDITOR_DATA = "EDITOR";
+		private final TreeItem currentItem;
+
+		protected TreeEditorDisposeListener(TreeItem currentItem) {
+			this.currentItem = currentItem;
+
+		}
+		public void widgetDisposed(DisposeEvent e) {
+			disposeTreeEditor();
+		}
+
+		protected void disposeTreeEditor() {
+			if (currentItem.getData(EDITOR_DATA) != null) {
+				TreeEditor editor = (TreeEditor) currentItem.getData(EDITOR_DATA);
+				editor.getEditor().dispose();
+				editor.dispose();
+			}
+		}
+		protected void removeDisposeListener() {
+			if (currentItem.getData(LISTENER_DATA) != null) {
+				currentItem.removeDisposeListener((DisposeListener) currentItem.getData(LISTENER_DATA));
+			}
+		}
+	}
+
+	/**
+	 * 
+	 * @author robert rudi - Initial contribution and API
+	 * 
+	 */
+	protected final static class NativeCellWidgetUtil {

```

Comment:
no static final, does not work with guice well

Reply:
Removed final